Oqtane.Server/Infrastructure/Jobs/HostedServiceBase.cs

```diff
@@ -153,27 +162,52 @@ public Task StartAsync(CancellationToken cancellationToken)
                 // set IsExecuting to false in case this job was forcefully terminated previously 
                 using (var scope = _serviceScopeFactory.CreateScope())
                 {
-                    string jobType = Utilities.GetFullTypeName(GetType().AssemblyQualifiedName);
+                    string jobTypeName = Utilities.GetFullTypeName(GetType().AssemblyQualifiedName);
                     IJobRepository jobs = scope.ServiceProvider.GetRequiredService<IJobRepository>();
-                    Job job = jobs.GetJobs().Where(item => item.JobType == jobType).FirstOrDefault();
+                    Job job = jobs.GetJobs().Where(item => item.JobType == jobTypeName).FirstOrDefault();
                     if (job != null)
                     {
                         job.IsStarted = true;
                         job.IsExecuting = false;
                         jobs.UpdateJob(job);
                     }
+                    else
+                    {
+                        // auto registration
+                        job = new Job { JobType = jobTypeName };
+                        // optional properties
+                        var jobType = Type.GetType(jobTypeName);
+                        var jobObject = ActivatorUtilities.CreateInstance(scope.ServiceProvider, jobType) as HostedServiceBase;
```

Comment:
Is it necessary to build new instance here to get defaults? this.Name must work.

Reply:
The StartAsync() method of IHostedService is only executed once when a Host starts up

Reply:
> 
> 
> The StartAsync() method of IHostedService is only executed once when a Host starts up
my fault, I do not recognize method where is it called. This is correct place.